storage/flux/reader.go

```diff
@@ -506,9 +506,35 @@ func defaultGroupKeyForSeries(tags models.Tags, bnds execute.Bounds) flux.GroupK
 	return execute.NewGroupKey(cols, vs)
 }
 
-func determineTableColsForGroup(tagKeys [][]byte, typ flux.ColType) ([]flux.ColMeta, [][]byte) {
-	cols := make([]flux.ColMeta, 4+len(tagKeys))
-	defs := make([][]byte, 4+len(tagKeys))
+func IsSelector(agg *datatypes.Aggregate) bool {
+	if agg == nil {
+		return false
+	}
+	return agg.Type == datatypes.AggregateTypeMin || agg.Type == datatypes.AggregateTypeMax ||
+		agg.Type == datatypes.AggregateTypeFirst || agg.Type == datatypes.AggregateTypeLast
+}
+
+func determineTableColsForGroup(tagKeys [][]byte, typ flux.ColType, agg *datatypes.Aggregate, groupKey flux.GroupKey) ([]flux.ColMeta, [][]byte) {
+	var colSize int
+	if agg == nil || IsSelector(agg) {
+		// The group without aggregate or with selector (min, max, first, last) case:
+		// _start, _stop, _time, _value + tags
+		colSize += 4 + len(tagKeys)
+	} else {
+		// The group aggregate case:
+		// Only the group keys + _value are needed.
+		// Note that `groupKey` will contain _start, _stop, plus any group columns specified.
```

Comment:
Why are _start and _stop always in the group key?

Reply:
I cannot speak to the design decision behind this, but this is how we currently implement it:
https://github.com/influxdata/influxdb/blob/master/storage/flux/reader.go#L539

Reply:
Don't they have to be removed from the group key at some point?

Reply:
I guess not? Did you see where we do that in the code?

Reply:
I'm looking for it, but I don't see it.